spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java

```diff
@@ -1061,6 +1080,30 @@ public String baz(Baz baz, String rk) {
 
 	}
 
+	@Target({ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE})
+	@Retention(RetentionPolicy.RUNTIME)
+	@RabbitListener(bindings = @QueueBinding(
+			value = @Queue,
+			exchange = @Exchange(value = "test.metaFanout", type = ExchangeTypes.FANOUT, autoDelete = "true")))
+	public @interface MyAnonFanoutListener {
```

Comment:
M-m-m. Looks like no. We do only this:
```
private Collection<RabbitListener> findListenerAnnotations(Method method) {
```
but don't collect attributes or `AnnotationUtils.synthesizeAnnotation()` from the custom environment. 

OK. That's fine too.

More complex cases should be achieved with the regular `@Bean` definition for the AMQP entities (exchanges, queues and bidnings between them) and used as a `queus` from the normal `@RabbitListener` annotation.

How about to document that as well?